### DISCUSSION TITLE
[BUG] Set-PnPFolderPermission: Object reference not set to an instance of an object on server. The object is associated with property ListItemAllFields.

### DISCUSSION THREAD

**Discussion by YuriySamorodov** [pnp/powershell#4148] [General]
### Reporting an Issue or Missing Feature
I am reporting a bug

### Expected behavior 
`Set-PnPFolderPermission -List Documents -User user@outlook.com -Identity $folder -AddRole Contribute` should add permission on a folder.
### Actual behavior 
Getting `Set-PnPFolderPermission: Object reference not set to an instance of an object on server. The object is associated with property ListItemAllFields.` error when trying to execute the command `Set-PnPFolderPermission -List Documents -User user@outlook.com -Identity $folder -AddRole Contribute`
![image](https://github.com/user-attachments/assets/4ef5ca74-77df-4ca0-a69f-693f5508fc75)

Related bugs: #4143 

### Steps to reproduce behavior
```
Connect-PnPOnline -Url https://tenant.sharepoint.com/sites/sitename
$list = Get-PnPList Documents
$folder = Get-PnPFolder -Url '/Shared Documents'
Set-PnPFolderPermission -List $list -User user@outlook.com -Identity $folder -AddRole Contribute
```
### What is the version of the Cmdlet module you are running?
PnP,PowerShell **2.5.39**

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@YuriySamorodov - Your `$list` and your `$folder` are essentially the same? Or did you create a folder called "Shared Documents"? This is confusing because the `Documents` library is `/Shared Documents` in the background.

 When I run the command with your values, I can reproduce the error. However, if you have created a folder yourself called "Shared Documents", then your `-Identity` should be "Documents/Shared Documents". 
Have a look at the examples: https://pnp.github.io/powershell/cmdlets/Set-PnPFolderPermission.html

I have no issue otherwise when trying with `2.5.42`.

**Comment by YuriySamorodov** [comment]
Hi, @veronicageek . Shared Documents is the default Documents library. `$list` and $folder are the same, yes.

**Comment by jackpoz** [comment]
I submitted a PR at https://github.com/pnp/powershell/pull/4157 to change the error message. [Set-PnPListPermission](https://pnp.github.io/powershell/cmdlets/Set-PnPListPermission.html) is probably what you should use to set permissions at list/library level.
